### PR TITLE
Add refresh button to process picker

### PR DESCRIPTION
**Pull request by mbise1993**
Issue [#2885](https://github.com/Microsoft/vscode-cpptools/issues/2885)

### CODE REVIEW

**Comment by sean-mcmanus** [comment]
Yeah, whenever I do the Attach too early and the process list is out of date, having a refresh option would be useful.

File: Extension/assets/Refresh_inverse.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16"><path d="M13.451 5.609l-.579-.939-1.068.812-.076.094c-.335.415-.927 1.341-1.124 2.876l-.021.165.033.163.071.345c0 1.654-1.346 3-3 3-.795 0-1.545-.311-2.107-.868-.563-.567-.873-1.317-.873-2.111 0-1.431 1.007-2.632 2.351-2.929v2.926s2.528-2.087 2.984-2.461h.012l3.061-2.582-4.919-4.1h-1.137v2.404c-3.429.318-6.121 3.211-6.121 6.721 0 1.809.707 3.508 1.986 4.782 1.277 1.282 2.976 1.988 4.784 1.988 3.722 0 6.75-3.028 6.75-6.75 0-1.245-.349-2.468-1.007-3.536z" fill="#2D2D30"/><path d="M12.6 6.134l-.094.071c-.269.333-.746 1.096-.91 2.375.057.277.092.495.092.545 0 2.206-1.794 4-4 4-1.098 0-2.093-.445-2.817-1.164-.718-.724-1.163-1.718-1.163-2.815 0-2.206 1.794-4 4-4l.351.025v1.85s1.626-1.342 1.631-1.339l1.869-1.577-3.5-2.917v2.218l-.371-.03c-3.176 0-5.75 2.574-5.75 5.75 0 1.593.648 3.034 1.695 4.076 1.042 1.046 2.482 1.694 4.076 1.694 3.176 0 5.75-2.574 5.75-5.75-.001-1.106-.318-2.135-.859-3.012z" fill="#C5C5C5"/></svg>
```
Comment (WardenGnaw) [inline Extension/assets/Refresh_inverse.svg:1]: Did you create this SVG or did you get it from somewhere?

Comment (mbise1993) [reply Extension/assets/Refresh_inverse.svg:1]: I got it from the icons included in VS Code

File: Extension/src/Debugger/attachQuickPick.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as util from '../common';
+import * as vscode from 'vscode';
+
+class RefreshButton implements vscode.QuickInputButton {
+    get iconPath(): vscode.Uri {
+        const refreshImagePath: string = util.getExtensionFilePath("assets/Refresh_inverse.svg");
```
Comment (sean-mcmanus) [inline Extension/src/Debugger/attachQuickPick.ts]: It seems like this won't appear correctly if the user has a non-dark theme. Do you know how to check if the non-inverse icon should be used?

Comment (WardenGnaw) [reply Extension/src/Debugger/attachQuickPick.ts]: iconPath can return `{ light: Uri; dark: Uri }`

File: Extension/src/Debugger/attachQuickPick.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    id: string;
+}
+
+export function showQuickPick(getAttachItems: () => Promise<AttachItem[]>): Promise<string> {
+    return getAttachItems().then(processEntries => {
+        return new Promise<string>((resolve, reject) => {
+            let quickPick: vscode.QuickPick<AttachItem> = vscode.window.createQuickPick<AttachItem>();
+            quickPick.canSelectMany = false;
+            quickPick.matchOnDescription = true;
+            quickPick.matchOnDetail = true;
```
Comment (WardenGnaw) [inline Extension/src/Debugger/attachQuickPick.ts:36]: Add `quickPick.title = "Attach to Process"`

The button alone makes it difficult for the user to see the refresh icon.

File: Extension/src/Debugger/attachQuickPick.ts (hunk 2 above)
Comment (WardenGnaw) [reply Extension/src/Debugger/attachQuickPick.ts]: Does not seem to be an issue in Light+ theme. 
image

Comment (mbise1993) [reply Extension/src/Debugger/attachQuickPick.ts]: I just pushed a change that uses different icons for dark and light themes.

File: Extension/assets/Refresh_inverse.svg (hunk 1 above)
Comment (WardenGnaw) [reply Extension/assets/Refresh_inverse.svg:1]: We are currently waiting to see if it is ok for our extension to use the icons. Once that is cleared, I will approve and complete the PR.

**Review by sean-mcmanus — APPROVED** [review]

**Review by WardenGnaw — APPROVED** [review]

**Comment by WardenGnaw** [comment]
@mbise1993 Thank you for your contribution to the cpptools-extension.

**Comment by mbise1993** [comment]
@WardenGnaw My pleasure!

**Pull request merged**